woodwork/table_schema.py

```diff
@@ -470,15 +470,15 @@ def _filter_cols(self, include=None, exclude=None, col_names=False):
             if col_names and selector in self.columns:
                 col_name_matches.add(selector)
 
-        cols_to_return = set()
+        cols_to_return = []
         for col_name, col in self.columns.items():
             is_match = (type(col.logical_type) in ltypes_used or
                         col.semantic_tags.intersection(tags_used) or
                         col_name in col_name_matches)
-            if include is not None and is_match:
-                cols_to_return.add(col_name)
-            elif exclude is not None and not is_match:
-                cols_to_return.add(col_name)
+            if include is not None and is_match and col_name not in cols_to_return:
```

Comment:
Mostly looks good.  Just one thought: not sure about the broader context of this but technically pandas does support columns with the same name.  Would doing this `not in` check here possibly cause issues with a dataset that has duplicate column names in the same table?

Reply:
@davesque Although pandas does support duplicate column names, Woodwork does not, as the column name serves as the key by which we track the typing information. Any duplicates that occur here would be a result of a single column matching multiple select criteria, not because duplicate columns exist in the dataframe.